src/runtime/print.go

```diff
@@ -372,3 +372,12 @@ func printbool(b bool) {
 		printstring("false")
 	}
 }
+
+func printslice(ptr uintptr, len_, cap_ int64) {
```

Comment:
`len` and `cap` are also of `uintptr` type. They just happen to match `int64` on a 64-bit system.

Reply:
I think I can use int in this case too to avoid converting it back to int? That should work in both 32 and 64 bit systems.

Reply:
There are also 16-bit systems: AVR. While `len(slice)` returns an int on AVR to comply with the spec, it is internally stored as a pointer-sized integer (which is 16 bits on AVR) and thus matches `uintptr`.
You can probably add a new `printuintptr` like this:

```go
func printuintptr(n uintptr) {
    switch unsafe.Sizeof(n) {
    case 2:
        printuint16(uint16(n))
    // etc
    }
}
```

You can test whether it works there by running `tinygo run -target=simavr ./testdata/print.go`.

Reply:
Thanks again for the cluebat. I will update it.

Reply:
This works, but it results in larger code than necessary on AVR. (Note that the switch on `unsafe.Sizeof` is resolved at compile time, not at runtime). So I'd really prefer a `printuintptr` that prints numbers based on the actual integer size.

Here is some code that also this, and can be factored out:

https://github.com/tinygo-org/tinygo/blob/b65447c7d567eea495805656f45472cc3c483e03/src/runtime/print.go#L326-L333

Reply:
Ok, I will update it accordingly.

Reply:
Done